dex/farm/src/base_functions.rs

```diff
@@ -304,15 +310,17 @@ where
         storage_cache: &mut StorageCache<Self::FarmSc>,
     ) {
         let total_reward = Self::mint_per_block_rewards(sc, &storage_cache.reward_token_id);
-        if total_reward > 0u64 {
-            storage_cache.reward_reserve += &total_reward;
-            let split_rewards = sc.take_reward_slice(total_reward);
-
-            if storage_cache.farm_token_supply != 0u64 {
-                let increase = (&split_rewards.base_farm * &storage_cache.division_safety_constant)
-                    / &storage_cache.farm_token_supply;
-                storage_cache.reward_per_share += &increase;
-            }
+        if total_reward == 0u64 {
+            return;
+        }
+
+        storage_cache.reward_reserve += &total_reward;
+        let split_rewards = sc.take_reward_slice(total_reward);
```

Comment:
I would add a check in the `startProduceRewards` endpoint, to not allow the rewards to start generating, if current_epoch < first_week_start_epoch. This way, we avoid getting to the point where we may call an endpoint (like `enterFarm`) that calls the `take_reward_slice` function, which then would throw a SC error.